src/shared/sysctl-util.c

```diff
@@ -53,14 +57,18 @@ char *sysctl_normalize(char *s) {
 
 int sysctl_write(const char *property, const char *value) {
         char *p;
+        _cleanup_close_ int fd = -1;
 
         assert(property);
         assert(value);
 
         log_debug("Setting '%s' to '%s'", property, value);
 
         p = strjoina("/proc/sys/", property);
-        return write_string_file(p, value, WRITE_STRING_FILE_DISABLE_BUFFER);
+        fd = open(p, O_WRONLY);
+        if (fd == -1)
+                return -errno;
+        return loop_write(fd, value, strlen(value), false);
```

Comment:
so this version does not append "\n" to the end if missing, like the old code did. I am pretty sure we should do that still, should we notice it is missing. After all sysctl is supposed to be used with shell commands such as "echo foo > /proc/sys/…", and those commands generally append a newline too. 

Moreover, there's no real need to use loop_write() here, as fs access is synchronous, and hence write() will never be return short on this. 

Also, I figure we should write the whole sysctl in one go, instead of multiple, hence maybe use code like this:

```c
if (!endswith(value, "\n"))
        value = strjoina(value, "\n");
if (write(fd, value, strlen(value)) < 0)
        return -errno;
return 0;
```